edit/location/location.go

```diff
@@ -149,10 +156,39 @@ func showPath(path, home string) string {
 
 func (p *provider) Filter(filter string) int {
 	p.filtered = nil
-	pattern := makeLocationFilterPattern(filter)
-	for _, item := range p.all {
-		if pattern.MatchString(showPath(item.Path, p.home)) {
-			p.filtered = append(p.filtered, item)
+
```

Comment:
This is mostly duplicate with `edit/completion/candidate.go`. It is an awkward one, not very easy to factor out.

But you can always document that fact. It can serve the interest of future contributors; at some point the evaler API might have changed in way to make this easier to factor out.

Reply:
Done.